src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -280,24 +268,19 @@ public void runDevAppServerCommand(List<String> args, Map<String, String> enviro
    * Uses the process runner to execute the classic Java SDK devappsever command.
    *
    * @param args the arguments to pass to devappserver
-   * @param environment map of environment variables to set for the dev_appserver process
    * @throws ProcessRunnerException when process runner encounters an error
    * @throws CloudSdkNotFoundException when the Cloud SDK is not installed where expected
    * @throws CloudSdkOutOfDateException when the installed Cloud SDK is too old 
    * @throws AppEngineException when dev appserver cannot be found
    */
-  public void runDevAppServer1Command(List<String> jvmArgs, List<String> args, 
-          Map<String, String> environment)
+  public void runDevAppServer1Command(List<String> jvmArgs, List<String> args)
           throws ProcessRunnerException {
     validateAppEngineJavaComponents();
+    validateJdk();
 
     List<String> command = new ArrayList<>();
-    String javaHome = environment.get("JAVA_HOME");
-    if (javaHome == null) {
-      javaHome = System.getProperty("java.home");
-    }
-    command.add(
-            Paths.get(javaHome).resolve("bin/java").toAbsolutePath().toString());
+
+    command.add(getJavaExecutablePath().toAbsolutePath().toString());
```

Comment:
Since getJavaExecutablePath() is already calling toAbsolutePath(), need to remove this from here.